src/dotnet-svcutil/lib/src/Shared/ProjectDependency.cs

```diff
@@ -28,6 +28,7 @@ internal class ProjectDependency : IComparable<ProjectDependency>
 
         private static readonly IReadOnlyList<string> s_binaryExtensions = new List<string> { ".exe", ".dll" }.AsReadOnly();
         private static readonly IReadOnlyList<string> s_projectExtensions = new List<string> { ".csproj" }.AsReadOnly();
+        private static readonly IReadOnlyList<string> s_exeExtensions = new List<string> { ".exe" }.AsReadOnly();
 
         /// <summary>
         /// The dependency name.  This can be a package or assembly name.
```

Comment:
similar to comments above, is this new variable really needed?

Reply:
In our case, this new variable **s_exeExtensions** defines EXE extension names we want to disallow as a package name ending, base on our discussion, **.exe** may have some side-effect so we still disallow it, and we exclude **.dll** extension which would be allowed for a package name end with it to address issue #4126. I think the new variable is necessary to differentiate with **s_binaryExtensions** which include both **.exe**  and **.dll.**